src/coreclr/vm/invokeutil.cpp

```diff
@@ -905,10 +907,17 @@ void InvokeUtil::SetValidField(CorElementType fldType,
         MethodTable *pMT = fldTH.AsMethodTable();
         {
             void* pFieldData;
+            fldTH.AsMethodTable()->EnsureActive();
+            obj = fldTH.AsMethodTable()->Allocate();
+            GCPROTECT_BEGIN(obj);
             if (pField->IsStatic())
                 pFieldData = pField->GetCurrentStaticAddress();
             else
-                pFieldData = (*((BYTE**)target)) + pField->GetOffset() + sizeof(Object);
+            {
+                OBJECTREF objRef = ObjectToOBJECTREF(obj);
+                pFieldData = pField->GetAddress(OBJECTREFToObject(objRef));
```

Comment:
I think we want the address of the target. I would have assumed we pass in the `target` here to get the address relative to that.